src/lambda/commands/downloadLambda.ts

```diff
@@ -234,3 +238,49 @@ function computeLambdaRoot(lambdaLocation: string, functionNode: LambdaFunctionN
 
     return lambdaIndex > -1 ? normalizedLocation.slice(0, lambdaIndex) : path.dirname(normalizedLocation)
 }
+
+export type ApplicationJson = {
+    ApplicationName?: string
+    DeploymentMethod: 'lambda' | 'cloudformation'
+    Functions: {
+        [logicalId: string]: {
+            PhysicalId: {
+                [region: string]: string
+            }
+        }
+    }
+    StackName?: string
+}
+
+export function createApplicationJson(lambdaFunc: Lambda.GetFunctionResponse, region: string): ApplicationJson {
+    const physicalId =
+        lambdaFunc.Configuration?.FunctionName !== undefined ? lambdaFunc.Configuration?.FunctionName : ''
+    const logicalId = lambdaFunc.Tags?.['aws:cloudformation:logical-id']
+        ? lambdaFunc.Tags['aws:cloudformation:logical-id']
+        : physicalId
+    const stackName = lambdaFunc.Tags?.['aws:cloudformation:stack-name']
+        ? lambdaFunc.Tags['aws:cloudformation:stack-name']
+        : undefined
+    const deploymentMethod = stackName !== undefined ? 'cloudformation' : 'lambda'
+    return {
+        DeploymentMethod: deploymentMethod,
+        Functions: {
+            [logicalId]: {
+                PhysicalId: {
+                    [region]: physicalId,
+                },
+            },
+        },
+        StackName: stackName,
+    }
+}
+
+function writeAppJsonFile(destinationPath: string, appJson: ApplicationJson) {
+    // Should not block download lambda if there is an error writing .application.json file
+    try {
+        fs.writeFile(path.join(destinationPath, '.application.json'), JSON.stringify(appJson))
+    } catch (e) {
+        const err = e as Error
+        getLogger().error('download lambda: Error writing .application.json file, %O', err)
```

Comment:
if the file fails to write, that is probably a failure that is worth surfacing to the user. So we should also show an error message here.

Also, our code guidelines mention that we generally want to avoid writing a stacktrace in the logs, it's not useful usually.

```suggestion
        getLogger().error('download lambda: Error writing .application.json file: %s', err.message)
```